### PR TITLE
Fixing issues with offset and limit where you cannot use offset and limit

### PR DESCRIPTION
**Pull request by trande4884**
Fixing issues with offset and limit where you cannot use offset and limit together as offset was being hardcoded to 0.

# Description

Making updates so that offset is honored with limit. I also added the required unit tests to cover the changes.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java [hunk 1]
```diff
@@ -72,15 +74,37 @@ public int getLimit() {
...
+    }
+
     /**
      * To set limit number
      *
      * @param limit int
      */
-    public void setLimit(int limit) {
+    public void withLimit(int limit) {
+        if (this.limit == 0) {
```
Comment (jialigit) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java:92]: what is this if(this.limit == 0) used for?

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java [hunk 2]
```diff
@@ -72,15 +74,37 @@ public int getLimit() {
...
+        return offset;
+    }
+
     /**
      * To set limit number
      *
      * @param limit int
      */
-    public void setLimit(int limit) {
+    public void withLimit(int limit) {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java:91]: We cannot change the existing API. The correct way to do this is to deprecate the existing API, (keep it - with an annotation), and add the new API.

**Review by kushagraThapar — CHANGES_REQUESTED** [review]

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java (hunk 2 above)
Comment (trande4884) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java:103]: Done :)

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java [hunk 3]
```diff
@@ -72,17 +74,52 @@ public int getLimit() {
...
+     * To set limit number and offset number
+     *
+     * @param offset long
+     * @param limit int
+     */
+    public void withOffsetAndLimit(long offset, int limit) {
+        if (this.limit == 0) {
+            this.limit = limit;
+        }
+        this.offset = offset;
```
Comment (xinlian12) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java:120]: do we need to do the same checking as limit?  (can only set once?)

Comment (trande4884) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java]: Done!

**Pull request merged**